### PR TITLE
Admin Workspaces API

### PR DESCRIPTION
**Pull request by omarismail**
## Description

This PR adds the [Admin Workspaces API](https://www.terraform.io/docs/cloud/api/admin/workspaces.html).

The current implemented methods are List, Read, and Delete

## Testing plan

Run tests against an Terraform Enterprise Instance.

## Output from tests

```
$ TFE_TOKEN=<token> TFE_ADDRESS=<adrress-of-tfe-instance>  ENABLE_TFE=1 go test -v ./...  -timeout=30m
...
PASS
ok  	github.com/hashicorp/go-tfe	485.784s
?   	github.com/hashicorp/go-tfe/examples/organizations	[no test files]
?   	github.com/hashicorp/go-tfe/examples/workspaces	[no test files]

```

## External links

- [Asana](https://app.asana.com/0/1199201948575144/1199972599424003)
- [Admin Workspace API Docs](https://www.terraform.io/docs/cloud/api/admin/workspaces.html)

### CODE REVIEW

File: admin_workspace.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+type adminWorkspaces struct {
+	client *Client
+}
+
+// AdminWorkspaces represents a Terraform Enterprise admin workspace.
+type AdminWorkspace struct {
+	ID      string   `jsonapi:"primary,workspaces"`
+	Name    string   `jsonapi:"attr,name"`
+	Locked  bool     `jsonapi:"attr,locked"`
+	VCSRepo *VCSRepo `jsonapi:"attr,vcs-repo"`
```
Comment (chrisarcand) [inline admin_workspace.go:37]: It looks like the API only returns the `identifier` for the this and not all the fields of this type will be populated; should we use some other simple type besides `*VCSRepo` to denote this, or would we be boxing ourselves in a corner doing something like `map[string]interface{}`? Whatcha think?

**Review by chrisarcand — CHANGES_REQUESTED** [review]
Looks great! The `include` error is my only blocker.

File: errors.go [hunk 2]
```diff
@@ -12,6 +12,9 @@ var (
 	// ErrResourceNotFound is returned when a receiving a 404.
 	ErrResourceNotFound = errors.New("resource not found")
 
+	// ErrInvalidIncludeParam is returned when an 'include' parameter is invalid.
+	ErrInvalidIncludeParam = errors.New("bad request\n\nInvalid include parameter")
```
[the remark this answers is not in the document]
Comment (omarismail) [reply errors.go:16]: Very fair point. I'll move it over to the test

File: admin_workspace.go (hunk 1 above)
Comment (omarismail) [reply admin_workspace.go:37]: Good catch! Looking at the serializers in TFC, `vcs_repo` is only used once for all Admin endpoints, so I think just nesting the struct here would do. 

```
// AdminWorkspaces represents a Terraform Enterprise admin workspace.
type AdminWorkspace struct {
	ID      string `jsonapi:"primary,workspaces"`
	Name    string `jsonapi:"attr,name"`
	Locked  bool   `jsonapi:"attr,locked"`
	VCSRepo *struct {
		Identifier string `json:"identifier"`
	} `jsonapi:"attr,vcs-repo"`
}
```

**Review by chrisarcand — APPROVED** [review]
Looks great!! ✅ :shipit:

**Pull request merged**